sensing/autoware_downsample_filters/src/filter.cpp

```diff
@@ -0,0 +1,344 @@
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * $Id: filter.cpp 35876 2011-02-09 01:04:36Z rusu $
+ *
+ */
+
+#include "autoware/downsample_filters/filter.hpp"
+
+#include "autoware/downsample_filters/memory.hpp"
+
+#include <autoware_utils_tf/transform_listener.hpp>
+#include <pcl_ros/transforms.hpp>
+#include <tf2_eigen/tf2_eigen.hpp>
+
+#include <pcl/io/io.h>
+
+#include <memory>
+#include <set>
+#include <string>
+#include <utility>
+#include <vector>
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+autoware::downsample_filters::Filter::Filter(
+  const std::string & filter_name, const rclcpp::NodeOptions & options)
+: Node(filter_name, options)
+{
+  // Set parameters (moved from NodeletLazy onInit)
+  {
+    tf_input_frame_ = static_cast<std::string>(declare_parameter("input_frame", ""));
+    tf_output_frame_ = static_cast<std::string>(declare_parameter("output_frame", ""));
+    max_queue_size_ = static_cast<std::size_t>(declare_parameter("max_queue_size", 5));
+
+    RCLCPP_DEBUG_STREAM(
+      this->get_logger(),
+      "Filter (as Component) successfully created with the following parameters:"
+        << std::endl
+        << " - max_queue_size   : " << max_queue_size_);
+  }
+
+  // Set publisher
+  {
+    rclcpp::PublisherOptions pub_options;
+    pub_options.qos_overriding_options = rclcpp::QosOverridingOptions::with_default_policies();
+    pub_output_ = this->create_publisher<PointCloud2>(
+      "output", rclcpp::SensorDataQoS().keep_last(max_queue_size_), pub_options);
+  }
+
+  // TODO(sykwer): Change the corresponding node to subscribe to `faster_input_callback`
+  // each time a child class supports the faster version.
+  // When all the child classes support the faster version, this workaround is deleted.
+  std::set<std::string> supported_nodes = {"VoxelGridDownsampleFilter"};
+  auto callback = supported_nodes.find(filter_name) != supported_nodes.end()
+                    ? &Filter::faster_input_callback
+                    : &Filter::input_callback;
+
+  // Subscribe in an old fashion to input only (no filters)
+  // CAN'T use auto-type here.
+  std::function<void(const PointCloud2ConstPtr msg)> cb =
+    std::bind(callback, this, std::placeholders::_1);
+  sub_input_ = create_subscription<PointCloud2>(
+    "input", rclcpp::SensorDataQoS().keep_last(max_queue_size_), cb);
+
+  // Set tf_listener, tf_buffer.
+  setupTF();
```

Comment:
setupTF() seems to be just online code.
We can just replace this line with it's contents

```suggestion
    transform_listener_ = std::make_unique<autoware_utils_tf::TransformListener>(this);
```